tekton/publish-nightly.yaml

```diff
@@ -107,6 +86,35 @@ spec:
 
       echo $VERSION_TAG > "/builder/home/version"
 
+  - name: create-ko-yaml
+    image: busybox
+    command:
+      - /bin/sh
+    args:
+      - -ce
+      - |
+        set -e
+        set -x
+
+        cat <<EOF > /workspace/go/src/github.com/tektoncd/pipeline/.ko.yaml
+        # By default ko will build images on top of distroless
+        baseImageOverrides:
+          # Use the images we just built as base images
+          ${inputs.params.pathToProject}/${outputs.resources.builtCredsInitImage.url}: ${inputs.params.imageRegistry}/${inputs.params.pathToProject}/build-base:latest
+          ${inputs.params.pathToProject}/${outputs.resources.builtGitInitImage.url}: ${inputs.params.imageRegistry}/${inputs.params.pathToProject}/build-base:latest
+          ${inputs.params.pathToProject}/${outputs.resources.builtEntrypointImage.url}: busybox # image should have shell in $PATH
+        baseBuildOverrides:
+          $(input.params.pathToProject)/$(outputs.resources.builtControllerImage.url):
+            env:
+              - name: CGO_ENABLED
+                value: 1
+            flags:
+              - name: ldflags
+                value: "-X $(input.params.pathToProject)/pkg/version.PipelineVersion=$(cat /builder/home/version)"
```

Comment:
This is freaky :sweat_smile: :stuck_out_tongue_closed_eyes: the 2nd `$(…)` is bash, the first one is us :sweat_smile:

Reply:
Yeah, sometimes I define step env variables for the Tekton ones, to avoid the mix and match.